### PR TITLE
Fix standalone torchax converter for Qwen3.5, and enable ATTN_DP and EP

### PR DESCRIPTION
**Pull request by wenxindongwork**
# Description

Makes `use_standalone_converter` work end to end for Qwen3.5 against current tpu-inference, so RL rollouts run on vLLM's **native** Qwen3.5 (torchax) model with weight sync from MaxText.

- **`torchax_converter/qwen35_moe.py`**: updates the converter to the current tpu-inference runner layout (the drift behind #4713) — transposes on the fused projections (and none on `o_proj` / GDN `out_proj` / shared `down_proj`), KV-head replication when attention shards exceed KV heads, current + legacy routed-expert names, `A_log` kept float32.
- **`maxtext_vllm_rollout.py`**: actually applies the converter — `_create_model_converter` instantiates it when `use_standalone_converter` is set, and `MaxTextVllmSampler.update_params` syncs via a new `_sync_standalone_converted` (KV-cache teardown → convert → chunked reshard → in-place runner-state assignment → KV-cache rebuild).
- **EP + attention DP**: the converter is parameterized on the sampler's sharding (`attn_dp_size`, `enable_expert_parallel`, filled automatically from the rollout config): attention/GDN/shared-expert shard `tp // attn_dp` ways, and routed experts use the GMM_EP layout under expert parallelism. Defaults reproduce plain TP bit-for-bit.

To use in the e2e: set `use_standalone_converter=True` and remove the `MaxTextForCausalLM` hf_overrides + `maxtext_config` so vLLM runs its native model.

Caveat: this mirrors tpu-inference's internal layout, so it must be revalidated when tpu-inference changes weight processing, and FP8 checkpoints are not covered. The canonical-layout path (#5034 + vllm-project/tpu-inference#3477 + google/tunix#2014) removes that coupling.

# Tests

- Unit tests (`tests/post_training/unit/qwen35_standalone_converter_test.py`, 25 cases): the real `convert()` on a small synthetic scanned state in all three sharding modes (plain TP, attention DP, EP), checked against expectations computed directly from the inputs — KV-head replication, GMM_TP/GMM_EP expert layouts, untransposed kernels, dtypes, expert-name aliases, full key coverage — plus the converter selection/hint plumbing and the `_sync_standalone_converted` paths (routing, in-place update, chunked/fallback reshard, layout-drift and non-dict-state errors) with a mocked runner.

- gsm8k **0.9621/0.9629** (`lm_eval`, TP=8 + `--enable-expert-parallel` + `attn_dp_size=4`); the synced state is bit-identical to the HF-loaded state that scored it.
- Live e2e on TPU (v7x-8): MaxText ckpt → sampler sync into a running, randomized native vLLM engine (TP=8 + attn_dp=4 + EP): all dumped tensors bit-identical to the HF-loaded reference, 4/4 greedy generations identical, sync 36 s.
- CPU bit-validation of `convert()` against runner-state dumps in both plain TP=8 (all 573 tensors) and TP=8 + attn_dp=4 + EP; coverage check confirms every runner weight is produced.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run end-to-end tests tests and provided workload links above if applicable.
- [x] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

**Review by YixuanWang-99 — APPROVED** [review]

File: src/maxtext/integration/vllm/maxtext_vllm_rollout.py [hunk 1]
```diff
@@ -73,10 +75,27 @@ def _create_model_converter(
     mesh: jax.sharding.Mesh,
     use_hf_mapping: bool = False,
     use_weight_converter: bool = False,
+    use_standalone_converter: bool = False,
+    sharding_hints: Optional[dict] = None,
```
Comment (hengtaoguo) [inline src/maxtext/integration/vllm/maxtext_vllm_rollout.py:79]: What does this `sharding_hints` contain/mirror from the tpu-inference? Is it allowing any customized keys?

**Review by hengtaoguo — APPROVED** [review]
Thanks for the update fix!

File: src/maxtext/integration/vllm/maxtext_vllm_rollout.py (hunk 1 above)
Comment (wenxindongwork) [reply src/maxtext/integration/vllm/maxtext_vllm_rollout.py:79]: thanks for the thorough review. currently sharding_hints contains only two params

```
sharding_hints = {
    "attn_dp_size":     int
    "enable_expert_parallel":  True/False
}
```
